src/configuration.c

```diff
@@ -543,6 +544,21 @@ config_param_val_long(char *str, long *dst, int non_negative)
 	return (1);
 }
 
+static int
+config_param_val_workers(char *str, long *dst, int non_negative)
+{
+	assert(str != NULL);
+
+#if SC_NPROCESSORS_ONLN_WORKS
+	if (strcasecmp(str, "auto") == 0) {
+		/* Get number of active CPU */
+		*dst = sysconf(_SC_NPROCESSORS_ONLN);
+		return (1);
+	}
+#endif
```

Comment:
The `#if` should be inside the `if` so that "auto" is always checked, then we can have an `#else` to fall "auto" back to 1, possibly with a warning.